docs/spec/callables.rst

```diff
@@ -3,47 +3,107 @@
 Callables
 =========
 
-Argument defaults
------------------
+Terminology
+-----------
+
+In this section, and throughout this specification, the term  "parameter"
+refers to a named symbol associated with a function that accepts the value of
+an argument (or multiple arguments) passed to the function. The term
+"argument" refers to a value passed to a function when it is called.
+
+Python supports five kinds of parameters: positional-only, keyword-only,
+standard (positional or keyword), variadic positional (``*args``), and
+variadic keyword (``**kwargs``). Positional-only parameters can accept only
+positional arguments, and keyword-only parameters can accept only keyword
+arguments. Standard parameters can accept either positional or keyword
+arguments. Parameters of the form ``*args`` and ``**kwargs`` are variadic
+and accept zero or more positional or keyword arguments, respectively.
+
+In the example below, ``a`` is a positional-only parameter, ``b`` is
+a standard (positional or keyword) parameter, ``c`` is a keyword-only parameter,
+``args`` is a variadic parameter that accepts additional positional arguments,
+and ``kwargs`` is a variadic parameter that accepts additional keyword
+arguments::
+
+    def func(a: str, /, b, *args, c=0, **kwargs) -> None:
+        ...
+
+A function's "signature" refers to its list of parameters (including
+the name, kind, optional declared type, and whether it has a default
+argument value) plus its return type. The signature of the function above is
+``(a: str, /, b, *args, c=..., **kwargs) -> None``. Note that the default
+argument value for parameter ``c`` is denoted as ``...`` here because the
+presence of a default value is considered part of the signature, but the
+specific value is not.
+
+The term "input signature" is used to refer to only the inputs of a function.
+In the example above, the input signature is ``(a: str, /, b, *args, c=..., **kwargs)``.
+
+Positional-only parameters
+--------------------------
+
+Within a function signature, positional-only parameters are separated from
+non-positional-only parameters by a single forward slash ('/'). This
+forward slash does not represent a parameter, but rather a delimiter. In this
+example, ``a`` is a positional-only parameter and ``b`` is a standard
+(positional or keyword) parameter::
+
+    def func(a: int, /, b: int) -> None:
+        ...
 
-It may be useful to declare an argument as having a default
-without specifying the actual default value.  For example::
+    func(1, 2)  # OK
+    func(1, b=2)  # OK
+    func(a=1, b=2)  # Error
 
-  def foo(x: AnyStr, y: AnyStr = ...) -> AnyStr: ...
+Support for the ``/`` delimiter was introduced in Python 3.8 (:pep:`570`).
+Prior to Python 3.8, positional parameters were specified using a :ref:`naming
+convention <pos-only-double-underscore>`.
```

Comment:
```suggestion
For compatibility with earlier versions of Python, the type system also
supports specifying positional-only parameters using a :ref:`naming
convention <pos-only-double-underscore>`.
```
Nitpick: The convention is still supported, and the convention is not part of Python itself but only of the type system.